src/libraries/System.Runtime/tests/System.Runtime.Tests/System/AttributeTests.cs

```diff
@@ -167,7 +167,7 @@ public static void StringArgument_InvalidCodeUnits_FallbackUsed()
 
             string stringArg = cad.ConstructorArguments[0].Value as string;
             Assert.NotNull(stringArg);
-            Assert.Equal("\uFFFD\uFFFD", stringArg);
+            Assert.Equal("\uFFFD\uFFFD\uFFFD", stringArg);
```

Comment:
This is failing on Mono now.

Reply:
Might be related to https://github.com/dotnet/runtime/commit/beeb5e009fec205e334faaddc98944b522c2cd03#diff-9bd9d0e46241d07dc62110246c89c9f056333eff4b0b089551780a625c871d0d ?

https://github.com/dotnet/runtime/blob/135fec006e727a31763271984cd712f1659ccbd3/src/mono/mono/metadata/custom-attrs.c#L414-L419

Reply:
Thanks for the tip @lambdageek, that does look promising.

Reply:
Sad. yes, this is the offending call, but it is doing the right thing and calling into the native utf8.c implementation. The native implementation matches the Win32 `MultiByteToWideChar()` API, yay, but fails differently than the managed UTF-8 parser, the WTF-8 behavior that is being alluded to here is the issue.

I am using `Encoding.UTF8.GetString()` in the managed impl so perhaps I need to use something else instead. I will start there.

Reply:
I do not think we care about the exact failure more for invalid UTF8 strings in the binaries. The behavior is not tested in general and I do not expect it to be consistent.

Reply:
Yay, thank you for letting me know.